### PR TITLE
🍽 Add table directive

### PR DESCRIPTION
**Pull request by rowanc1**
See #593

### CODE REVIEW

File: packages/myst-directives/src/table.ts [hunk 1]
```diff
@@ -2,6 +2,54 @@ import type { DirectiveSpec, DirectiveData, GenericNode } from 'myst-common';
...
+  },
+  run(data: DirectiveData): GenericNode[] {
+    const children = [];
+    if (data.arg) {
+      children.push({
+        type: 'caption',
+        children: [{ type: 'paragraph', children: data.arg as GenericNode[] }],
+      });
+    }
+    children.push(...(data.body as GenericNode[]));
```
Comment (fwkoch) [inline packages/myst-directives/src/table.ts:39]: Could add validation here that children includes a `table` node? 🤷‍♀️ Trade-off between strictness now vs. unexpected behaviour later.

Looks like in other directives we err towards less strict, so maybe best to leave as-is.

**Review by fwkoch — APPROVED** [review]

File: packages/myst-directives/src/table.ts (hunk 1 above)
Comment (rowanc1) [reply packages/myst-directives/src/table.ts:39]: I did think about that as well. Probably a good idea to add it and at least warn.

**Comment by rowanc1** [comment]
@fwkoch to support better error messages I had to include `node` in the directive/role data. This allows us to raise specific errors about the directive.

This includes errors about options, which now point directly to the line where the error occurred:
image

**Comment by fwkoch** [comment]
Cool, yeah this is great! Gives the `run` function more power (I don't think there's any drawback to that - for the most part, using validated options/args/body in the run function will be easier and preferable).

Hmm, you know, we also have the `validate` function which now gets the `node` as well... worth considering if separating `run` and `validate` is actually useful or if we should just keep the extra validation inside the `run` function as we do everywhere, including this PR. (And just delete `validate` entirely)

**Comment by rowanc1** [comment]
I was thinking that as well, the only place we are using validate is in `list-table`, might be overkill. I suppose there may be a reason to validate without transforming? If you think we should remove this .. can we open another PR to do it?

Going to get this one in! 🚀

**Pull request merged**